jax/interpreters/partial_eval.py

```diff
@@ -483,14 +496,17 @@ def new_eqn_recipe(invars, outvars, primitive, params):
                         params)
 
 
-def recipe_to_eqn(unused_var, getvar, recipe):
+def recipe_to_eqn(unused_var: Callable[[], core.Var],
+                  getvar: Callable[[JaxprTracer], core.Var],
+                  recipe: JaxprEqnRecipe) -> core.JaxprEqn:
   _, in_tracers, out_tracer_refs, primitive, params = recipe
   out_tracers = [t_ref() for t_ref in out_tracer_refs]
   invars  = [getvar(t) for t in in_tracers]
   outvars = [unused_var() if t is None else getvar(t) for t in out_tracers]
   return new_jaxpr_eqn(invars, outvars, primitive, params)
 
-def tracers_to_jaxpr(in_tracers, out_tracers):
+def tracers_to_jaxpr(in_tracers: List[JaxprTracer],
+                     out_tracers: List[JaxprTracer]) -> Tuple[Jaxpr, Any, Any]:
```

Comment:
The last two elements of the result are better described as `Sequence[core.Var]`.

Reply:
Actually I think they are `Sequence[Any]` because they are const vals and env vals, rathe than the variables (which are stored in the jaxpr itself).